### DISCUSSION TITLE
What should the message __repr__ and __str__ methods look like?

### DISCUSSION THREAD

**Discussion by jmw182** [pitt-rnel/pylsb#31] [General]
In Dave's pull request, I pushed a commit that added __repr__ methods to the message and message data classes. This _works_ (so I'm not opening an issue), but may not be ideal and may warrant further discussion.

I also am creating this discussion to save a few stackoverflow tabs I have had open that may help improve the existing method:
https://stackoverflow.com/questions/20986330/print-all-fields-of-ctypes-structure-with-introspection
https://stackoverflow.com/questions/26927685/introspecting-nested-ctypes-structures

**Comment by dmweir** [comment]
I like the idea of creating a MessageData class. For now, this would just wrap some ctypes functions, but could optionally use other representations under the hood in the future.  The base class could have methods that the user could define to decide the proper string representation. That way we don't make any assumptions in ```__str__ ```requiring ctypes.  For example, we could make a subclasses called CtypesData or RTMAData that would use something similar to what you implemented now.
